src/Dfc.CourseDirectory.Core/BinaryStorageProvider/IBinaryStorageProvider.cs

```diff
@@ -7,5 +8,6 @@ public interface IBinaryStorageProvider
     {
         Task<bool> TryDownloadFile(string path, Stream destination);
         Task UploadFile(string path, Stream source);
+        Task<IEnumerable<BlobFileInfo>> ListFiles(string path);
```

Comment:
```suggestion
        Task<IReadOnlyList<BlobFileInfo>> ListFiles(string path);
```

Reply:
Any particular reason?

Reply:
Means `Count` is more efficient and prevents having to protect against double enumeration

Reply:
> Means Count is more efficient

`Count` doesn't exist on `IEnumerable<T>`, `Count()` does but would require an enumeration
> prevents having to protect against double enumeration

I like this argument 👍